src/lib/storage/table.cpp

```diff
@@ -154,55 +134,36 @@ const ProxyChunk Table::get_chunk_with_access_counting(ChunkID chunk_id) const {
   return ProxyChunk(_chunks[chunk_id]);
 }
 
-void Table::emplace_chunk(const std::shared_ptr<Chunk>& chunk) {
-  if (_chunks.size() == 1 && (_chunks.back()->column_count() == 0 || _chunks.back()->size() == 0)) {
-    // the initial chunk was not used yet
-    _chunks.clear();
-  }
-  DebugAssert(chunk->column_count() > 0, "Trying to add chunk without columns.");
-  DebugAssert(chunk->column_count() == column_count(),
-              std::string("adding chunk with ") + std::to_string(chunk->column_count()) + " columns to table with " +
-                  std::to_string(column_count()) + " columns");
-  _chunks.emplace_back(chunk);
-}
-
-std::unique_lock<std::mutex> Table::acquire_append_mutex() { return std::unique_lock<std::mutex>(*_append_mutex); }
-
-TableType Table::get_type() const {
-  // Cannot answer this if the table has no content
-  Assert(!_chunks.empty() && column_count() > 0, "Table has no content, can't specify type");
+void Table::append_chunk(const ChunkColumns& columns, const std::optional<PolymorphicAllocator<Chunk>>& alloc,
+                         const std::shared_ptr<ChunkAccessCounter>& access_counter) {
+  const auto chunk_size = columns.empty() ? 0u : columns[0]->size();
 
-  // We assume if one column is a reference column, all are.
-  const auto column = _chunks[0]->get_column(ColumnID{0});
-  const auto ref_column = std::dynamic_pointer_cast<const ReferenceColumn>(column);
-
-  if (ref_column != nullptr) {
-// In debug mode we're pedantic and check whether all columns in all chunks are Reference Columns
 #if IS_DEBUG
```

Comment:
I’d prefer 
```c++
DebugAssert([&]() {
  return true;
}(), "comment");
```

If you keep this style, please replace `DebugAssert` with `Assert` below.

Reply:
The current style makes it easier to debug, with a lambda you just know THAT it failed, not WHERE and in what state.